creation/create_cvmfsexec_distros.sh

```diff
@@ -3,123 +3,185 @@
-
-    declare -a machine_types
-    machine_types=($(echo $SUPPORTED_TYPES | tr ":" "\n"))
-
-    for cvmfs_src in "${cvmfs_sources[@]}"
-    do
-        for mach_type in "${machine_types[@]}"
-        do
-            echo -n "Making $cvmfs_src distribution for $mach_type machine..."
-            os=`echo $mach_type | awk -F'-' '{print $1}'`
-            arch=`echo $mach_type | awk -F'-' '{print $2}'`
-            if "$cvmfsexec_latest"/makedist -m $mach_type $cvmfs_src &> /dev/null ; then
-                "$cvmfsexec_latest"/makedist -o "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                if [[ -e "$cvmfsexec_distros"/cvmfsexec-${cvmfs_src}-${os}-${arch} ]]; then
-                    echo " Success"
-                    tar -cvzf "$cvmfsexec_tarballs"/cvmfsexec_${cvmfs_src}_${os}_${arch}.tar.gz -C "$cvmfsexec_distros" cvmfsexec-${cvmfs_src}-${os}-${arch} &> /dev/null
-                fi
-            else
-                echo " Failed! REASON: $cvmfs_src may not yet have a $mach_type build."
-            fi
-
-            # delete the dist directory within cvmfsexec to download the cvmfs configuration
-            # and repositories for another machine type
-            rm -rf "$cvmfsexec_latest"/dist
-        done
-    done
-
-    # remove the distros and latest folder under cvmfsexec.tmp
-    rm -rf "$cvmfsexec_distros"
-    rm -rf "$cvmfsexec_latest"
-fi
 
+####################### MAIN SCRIPT STARTS FROM HERE #######################
 
-# TODO: store/update version information in the $cvmfsexec_tarballs location for future reconfig/upgrade
-echo "$latest_ver" > "$cvmfsexec_tarballs"/.cvmfsexec_version
+# parsing the command-line arguments
+# check whether the first argument passed is the option '--work-dir'
+if [[ $1 == "--work-dir" ]]; then
+	# if value after this option is not a path (it is empty/unspecified), show usage documentation and exit
+	[[ ! -d "$2" ]] && usage || true
+	# otherwise, continue parsing the arguments
+	work_dir="$2"
+	shift 2
+else
+	# if --work-dir is not passed, assume default work-dir (RPM install)
+	# # check if there was a path specified without the --work-dir option, show usage documentation and exit
+	# [[ ! -d "$2" ]] && usage || true
+	work_dir="$DEFAULT_WORK_DIR"
+fi
 
-end=`date +%s`
+# check whether there are any remaining arguments passed to the script after the option
+if [[ $# -eq 0 ]]; then
+	echo "Building/Rebuilding of cvmfsexec distributions disabled!"
+	exit 0
+else
+	# if there are any arguments (max: 2, min: 1)
+	if [[ $# -eq 2 ]]; then
+		configurations=$1
+		machine_types=$2
+	elif [[ $# -eq 1 ]]; then
+		# if there is only one argument
+		arg_ending=${1:(-7)}
+		echo $arg_ending
+		if [[ $arg_ending == "-x86_64" || $arg_ending == "ppc64le" ]]; then
+			usage
+		else
+			# if false, SOURCES_LIST is empty, show usage and exit
+			configurations=$1; machine_types=$DEFAULT_MACHINE_TYPES
+		fi
+	else
+		echo "Invalid number of arguments passed!"
+		exit 1
+	fi
+fi
```

Comment:
Your script is actually accepting 0 arguments (considering a legal noop).
If the first argument is required you could use something like:
```shell
 if [[ $# -eq 0 || $# -gt 2 ]]; then
	echo "ERROR: Invalid number of arguments passed!"
    usage  # maybe remove the exit from usage if you'd like to have both exit 1 or 0 after it
	exit 1
fi
```
If you keep the noop (I lean towards this since it makes it easier to use):
```shell
 if [[ $# -eq 0 ]]; then
	echo "No sources specified. Building/Rebuilding of cvmfsexec distributions disabled!"
	exit 0
elif [[ $# -gt 2 ]]; then
	echo "ERROR: Invalid number of arguments passed!"
    usage  # maybe remove the exit from usage if you'd like to have both exit 1 or 0 after it
	exit 1
fi
```
After either one you are assured to have 1 or 2 arguments.
```shell
configurations=$1
machine_types=$(2:-$DEFAULT_MACHINE_TYPES}
```
If you control `$1` do it all the time. You can choose a positive or negative check. Or you can let later in the program do the check (or let makedist fail for invalid values). A typo in the name is probably more likely than the wrong order. If you check you'll have to update the script when the values change.
```
arg_ending=${1:(-7)}
# echo $arg_ending
if [[ $arg_ending == "-x86_64" || $arg_ending == "ppc64le" ]]; then
	echo "ERROR: you added a platform (...$arg_ending) where a source is required"
    usage
    exit 1
fi
```
After this the arguments should be OK

Reply:
updated as per this and the clarification discussion we had this week.